arm9/overlays/24/src/ov24_02254840.c

```diff
@@ -0,0 +1,254 @@
+    FreeToHeap(appHandler);
+}
+
+void ov24_02254918(void* arg0, MemoPadAppHandler* appHandler) {
+    static BOOL (*const ov24_02255100[3])(MemoPadAppHandler*) = {
+        ov24_022549AC, ov24_022549F8, ov24_02254C64
+    };
+    if (appHandler->unk0 < 3) {
+        ov20_02252C14(appHandler->unk16F8, appHandler->unk16FC);
+        if (ov24_02255100[appHandler->unk0](appHandler)) {
+            ov24_022548F4(appHandler);
+            sub_0200CAB4((s32)arg0);
+            ov20_022529A0(appHandler->unk16F8);
+        }
+    }
+}
+
+void ov24_02254960(int arg0, int arg1, int arg2, MemoPadAppHandler* appHandler) {
+    if (arg2 == 1) {
+        if (
+            (appHandler->drawState.stylusType == STYLUS_TYPE_DRAW && arg0 == 0)
+            || (appHandler->drawState.stylusType == STYLUS_TYPE_ERASE && arg0 == 1)
+        ) {
+            appHandler->drawState.stylusType ^= 1;
+            ov24_02254D8C(appHandler->displayHandler, 1);
+        }
+    }
+}
+
+void ov24_02254990(MemoPadAppHandler* appHandler) {
+    appHandler->unk2 = 1;
+}
+
+void ov24_02254998(MemoPadAppHandler* appHandler, u8 arg1) {
+    if (appHandler->unk2 == 0) {
+        appHandler->unk0 = arg1;
+    }
+    else {
+        appHandler->unk0 = 2;
+    }
+    appHandler->unk1 = 0;
+}
+
+BOOL ov24_022549AC(MemoPadAppHandler* appHandler) {
+    switch (appHandler->unk1) {
+        case 0:
+            ov24_02254D8C(appHandler->displayHandler, 0);
+            appHandler->unk1++;
+            break;
+        case 1:
+            if (ov24_02254DB0(appHandler->displayHandler, 0)) {
+                ov20_0225298C(appHandler->unk16F8);
+                ov24_02254998(appHandler, 1);
+            }
+            break;
+        default:
+            break;
+    }
+    return FALSE;
+}
+
+BOOL ov24_022549F8(MemoPadAppHandler* appHandler) {
+    if (appHandler->unk2 != 0) {
+        ov24_02254998(appHandler, 2);
+    }
+    switch (appHandler->unk1) {
+    case 0:
+        if (ov20_02252C08(appHandler->unk16F8)) {
+            break;
+        }
+        if (appHandler->unk3) {
+            u32 x = appHandler->drawState.lastModifiedX;
+            u32 y = appHandler->drawState.lastModifiedY;
+            if (ov24_02254AD4(appHandler)) {
+                ov24_02254B20(appHandler, x, y, appHandler->drawState.lastModifiedX, appHandler->drawState.lastModifiedY);
+            } else {
+                appHandler->unk3 = 0;
+            }
+        } else {
+            if (ov24_02254A70(appHandler)) {
+                ov24_02254D8C(appHandler->displayHandler, 3);
+                appHandler->unk3 = 1;
+            }
+        }
+        break;
+    }
+    return FALSE;
+}
+
+BOOL ov24_02254A70(MemoPadAppHandler* appHandler) {
+    u32 x, y;
+    if (TouchScreen_GetTapState(&x, &y)) {
+        if (((x - 16) < 156) & ((y - 16) < 150)) {
+            x = (x - 16) >> 1;
+            y = (y - 16) >> 1;
+            if (appHandler->drawState.pixelData[x][y] != appHandler->drawState.stylusType) {
```

Comment:
is stylusType the right name for the variable? this seems like a strange comparison/assignment

Reply:
I would think so, it's swapped based on whether the eraser or the pencil is selected in the gui and the comparison is basically just checking if the pixel data needs to be updated when the screen is tapped at its location.

Reply:
same here with touchType